### ISSUE TITLE
pre-trained model test

### ISSUE THREAD

**Issue by LiGengWan** [JustlfC03/SCUNet-plusplus#7]
Hi 
Thank you for your nice work.
I have a problem when I test your model in my private data.
I set the config path and load the pre-trained model in this way:

    net = ViT_seg(config, img_size=224, num_classes=2).cuda()
    net.load_from(config)

I inference my CTPA volume slice by slice, but the result seems not right.
I want to confirm:
1. if the model is loaded correctly.
2. the method of data preprocessing, such as normalization.

Thank you!

**Comment by satwiksunnam19** [comment]
Hello i've two issues
1. with the minimal changes to the  proposed architecture and the training part go well and when it comes to testing the model it sucks with the model loading part of dimensionality error. 

2. The Dataset part as i've seen that new FUMPE dataset and it's not in .npz format and anyone can help me with the part of code for conversion to the proposed format.


For the Issue 1 the problem ariases due to the below part of code 

`def forward_up_features(self, x, x_downsample):
        for inx, layer_up in enumerate(self.layers_up):
            if inx == 0:
                x = layer_up(x)
            else:
                skip = x_downsample[3 - inx]
                # Reshape for MHA
                B, L, C = x.shape
                H, W = int(math.sqrt(L)), int(math.sqrt(L))
                x = x.view(B, H, W, C).permute(0, 3, 1, 2)
                skip = skip.view(B, H, W, C).permute(0, 3, 1, 2)

                # Adjust dimensions for MHA
                x_mha = x.flatten(2).permute(2, 0, 1)  # Shape: [H*W, B, C]
                skip_mha = skip.flatten(2).permute(2, 0, 1)  # Shape: [H*W, B, C]

                # Dynamically create a MultiheadAttention layer with the correct embedding dimension
                mha = nn.MultiheadAttention(embed_dim=C, num_heads=self.num_heads[inx], batch_first=True)
                mha.to(x.device)
                # Combine features using MHA
                attn_output, _ = mha(query=x_mha, key=skip_mha, value=skip_mha)

                # Reshape back
                attn_output = attn_output.permute(1, 2, 0).view(B, C, H, W).permute(0, 2, 3, 1)
                x = attn_output.flatten(1, 2)
                # Dynamically adjust the input dimension of the linear layer
                next_layer_dim = int(self.embed_dim * 2 ** (self.num_layers - 1 - inx))
                if self.concat_back_dim[inx].in_features != C:
                    self.concat_back_dim[inx] = nn.Linear(C, next_layer_dim).to(x.device)
                x = self.concat_back_dim[inx](x)
                x = layer_up(x)
        x = self.norm_up(x)
        return x
`
@JustlfC03 @LiGengWan can anyone of help me with this
